client/src/components/modal.tsx

```diff
@@ -0,0 +1,162 @@
+import axios from 'axios';
+import React, { ChangeEvent, FormEvent, useRef, useState } from 'react';
+import { IAppointment } from '../../../shared/interfaces/appointment.interface';
+
+const Modal: React.FC = () => {
+    const [showModal, setShowModal] = useState(false);
+    const [error, setError] = useState('');
```

Comment:
This should probably be an object, as the response form the server will be an object if there's multiple validation issues.

Reply:
changed type to object. resolving this convo